src/EntityFramework.Core/ChangeTracking/Internal/StateManager.cs

```
@@ -85,6 +85,11 @@ public virtual InternalEntityEntry GetOrCreateEntry(object entity)
 
                 var entityType = _model.FindEntityType(entity.GetType());
 
+                if (entityType == null)
+                {
+                    throw new ModelItemNotFoundException(CoreStrings.EntityTypeNotFound(entity.GetType().DisplayName(false)));
```

I think we can just throw InvalidOperationException here. Pretty sure we decided in API review to get rid of the ModelItemNotFoundException and I don't think we want to bring it back.